failed to install

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "natural"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! webworker-threads@0.7.11 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the webworker-threads@0.7.11 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the webworker-threads package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs webworker-threads
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls webworker-threads
npm ERR! There is likely additional logging output above.

--- a/lib/natural/classifiers/classifier.js
+++ b/lib/natural/classifiers/classifier.js
@@ -21,10 +21,22 @@ THE SOFTWARE.
 */
 
 var PorterStemmer = require('../stemmers/porter_stemmer'),
-Threads = require('webworker-threads'),
 util = require('util'),
 events = require('events')
 os = require('os');
+
+try {
+    var Threads = require('webworker-threads');
+} catch (e) {
+    // Since webworker-threads are optional, only thow if the module is found
+    if (e.code !== 'MODULE_NOT_FOUND') throw e;
+}
+
+function checkThreadSupport() {
+    if (typeof Threads === 'undefined') {
+        throw new Error('parallel classification requires the optional dependency webworker-threads');
+    }
+}
 
 var Classifier = function(classifier, stemmer) {
     this.classifier = classifier;
@@ -142,6 +154,8 @@ function train() {
 }
 
 function trainParallel(numThreads, callback) {
+    checkThreadSupport();
+
     if (!callback) {
         callback = numThreads;
         numThreads = undefined;
@@ -238,6 +252,8 @@ function trainParallel(numThreads, callback) {
 }
 
 function trainParallelBatches(options) {
+    checkThreadSupport();
+
     var numThreads = options && options.numThreads;
     var batchSize = options && options.batchSize;
 

--- a/package.json
+++ b/package.json
@@ -13,13 +13,16 @@
   "dependencies": {
     "apparatus": ">= 0.0.9",
     "sylvester": ">= 0.0.12",
-    "underscore": ">=1.3.1",
-    "webworker-threads": ">=0.6.2"
+    "underscore": ">=1.3.1"
   },
   "devDependencies": {
-    "uubench": "0.0.x",
+    "jasmine-node": "~1.13.1",
+    "proxyquire": "^1.8.0",
     "sinon": "^1.12.2",
-    "jasmine-node": "~1.13.1"
+    "uubench": "0.0.x"
+  },
+  "optionalDependencies": {
+    "webworker-threads": ">=0.6.2"
   },
   "scripts": {
     "test": "NODE_PATH=. node_modules/jasmine-node/bin/jasmine-node spec/"

--- a/spec/missing_webworker_threads_spec.js
+++ b/spec/missing_webworker_threads_spec.js
@@ -1,0 +1,48 @@
+/*
+Copyright (c) 2017, Ryan Witt
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+var sinon = require('sinon');
+var proxyquire = require('proxyquire');
+var baseClassifier = proxyquire('lib/natural/classifiers/classifier.js', {
+    'webworker-threads': null
+});
+
+describe('missing webworker-threads', function() {
+    describe('classifier', function() {
+        it('should refuse to classify with parallel training', function() {
+            var classifier = new baseClassifier();
+            classifier.addDocument(['foo', 'bar'], 'baz');
+            expect(function() {
+                classifier.trainParallel(2)
+            }).toThrow(new Error('parallel classification requires the optional dependency webworker-threads'));
+        });
+
+        it('should refuse to classify with parallel batched training', function() {
+            var classifier = new baseClassifier();
+            classifier.addDocument(['foo', 'bar'], 'baz');
+            classifier.addDocument(['fizz', 'buzz'], 'boo');
+            expect(function() {
+                classifier.trainParallelBatches({numThreads: 2, batchSize: 2});
+            }).toThrow(new Error('parallel classification requires the optional dependency webworker-threads'));
+        });
+    });
+});

